### PR TITLE
Whitespace in Choosing Colormaps tutorial plots

### PR DESCRIPTION
**Pull request by Ptrskay3**
## PR Summary
This should solve #19308. However, I'm not sure if setting figsize explicitly is the way to go.

## PR Checklist

- [N/A] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by jklymak — CHANGES_REQUESTED** [review]
You are closer but https://52169-1385122-gh.circle-artifacts.com/0/doc/build/html/tutorials/colors/colormaps.html#sphx-glr-tutorials-colors-colormaps-py

You probably need to recalculate `top` and `bottom` since those are in fractions of the figure size.  Or maybe easier, make top and bottom 0 and add an extra axes at the top (i.e subplots(nrows=nrows+1,...)) to put the title in as text rather than a title.

**Review by jklymak — APPROVED** [review]
This isn't pixel perfect in that the widths of the bars are all a bit different, but I still think its an improvement over the current state....  I'll leave here for a day or so in case anyone else wants to comment.

File: tutorials/colors/colormaps.py [hunk 1]
```diff
@@ -221,8 +223,8 @@ def plot_color_gradients(cmap_category, cmap_list, nrows):
         ax.set_axis_off()
 
 
-for cmap_category, cmap_list in cmaps.items():
-    plot_color_gradients(cmap_category, cmap_list, nrows)
+for i, (cmap_category, cmap_list) in enumerate(cmaps.items()):
+    plot_color_gradients(cmap_category, cmap_list, nrows[i])
```
Comment (QuLogic) [inline tutorials/colors/colormaps.py:227]:
```suggestion
for rows, (cmap_category, cmap_list) in zip(nrows, cmaps.items()):
    plot_color_gradients(cmap_category, cmap_list, rows)
```
but do we really need `nrows` now? Can we not just check the length of `cmap_list` in `plot_color_gradients`?

**Comment by jklymak** [comment]
Actually, there is already code to do this properly at https://matplotlib.org//gallery/color/colormap_reference.html  Suggest we just duplicate it rather than having two versions...

**Comment by tacaswell** [comment]
Thanks @Ptrskay3 !  Congratulations on your first Matplotlib PR :tada:  Hopefully we will hear from you again!

**Pull request merged**